packages/vendure-plugin-sort-by-popularity/src/sort.service.ts

```diff
@@ -1,12 +1,83 @@
 import { Injectable } from '@nestjs/common';
-
+import {
+  Collection,
+  CollectionService,
+  OrderLine,
+  Product,
+  RequestContext,
+  TransactionalConnection,
+} from '@vendure/core';
+import { Success } from '../../test/src/generated/admin-graphql';
+import { CollectionTreeNode } from './helpers';
 @Injectable()
 export class SortService {
-  async setProductPopularity() {
-    // Get orders from pas X months
-    // Combine variant quantities per product
-    // Normalize by total quantity and store as popularity score on product
-    // Get all products and products of child collections for each collection
-    // Use sum of products per collection as popularity score
+  constructor(
+    private connection: TransactionalConnection,
+    private collectionService: CollectionService
+  ) {}
+  async setProductPopularity(ctx: RequestContext): Promise<Success> {
+    const groupedOrderLines = await this.connection
+      .getRepository(ctx, OrderLine)
+      .createQueryBuilder('orderLine')
+      .select([
+        'count(product.id) as count',
+        'orderLine.productVariant',
+        'orderLine.order',
+      ])
+      .innerJoin('orderLine.productVariant', 'productVariant')
+      .addSelect([
+        'productVariant.deletedAt',
+        'productVariant.enabled',
+        'productVariant.id',
+      ])
+      .innerJoin('orderLine.order', '`order`')
```

Comment:
Do we need the backticks (\`) here? It would be cleaner to just use `'order'`

Reply:
I will remove those